rtos/ConditionVariable.h

```diff
@@ -0,0 +1,70 @@
+/* mbed Microcontroller Library
+ * Copyright (c) 2006-2012 ARM Limited
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy
+ * of this software and associated documentation files (the "Software"), to deal
+ * in the Software without restriction, including without limitation the rights
+ * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+ * copies of the Software, and to permit persons to whom the Software is
+ * furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in
+ * all copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+ * SOFTWARE.
+ */
+#ifndef CONDITIONVARIABLE_H
+#define CONDITIONVARIABLE_H
+
+#include <stdint.h>
+#include "cmsis_os.h"
+#include "rtos/Mutex.h"
+
+namespace rtos {
+/** \addtogroup rtos */
+/** @{*/
+
+class Waiter;
+
+/** The ConditionVariable class is used to wait for condition changes
+ * while being protected by a mutex.
+*/
+class ConditionVariable {
+public:
+    /** Create and Initialize a ConditionVariable object */
+    ConditionVariable(Mutex &mutex);
+
+    /** Wait for a notification
+      @param   millisec  timeout value or 0 in case of no time-out. (default: osWaitForever)
+      @return  true if woken from notification, false if timed out.
+     */
+    bool wait(uint32_t millisec=osWaitForever);
+
+    /** Notify one waiter on this condition variable that a condition changed.
+     */
+    void notify();
+
+    /** Notify all waiters on this condition variable that a condition changed.
+     *
+     */
+    void notify_all();
+
+    ~ConditionVariable();
+
+private:
+    void add_wait_list(Waiter * waiter);
+    void remove_wait_list(Waiter * waiter);
+    Mutex &_mutex;
+    Waiter *wait_list;
```

Comment:
Could you declare (but not define!) private copy constructor and private assignment operator here. 
That would disallow copy of the `ConditionVariable` class.